### ISSUE TITLE
[no-loss-of-precision] The rule reports false positives for values with numeric separators

### ISSUE THREAD

**Issue by merlinnot** [eslint/eslint#13346]
**Tell us about your environment**

* **ESLint Version:** v7.1.0
* **Node Version:** v14.0.0
* **npm Version:** v6.14.4

**What parser (default, Babel-ESLint, etc.) are you using?**

`@typescript-eslint/parser`

**Please show your full configuration:**

It's in a public repository: https://github.com/ridedott/eslint-config

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const x = 1_000;
```

```bash
eslint --cache --cache-location './.cache/eslint' --fix '**/*.ts'
```

**What did you expect to happen?**

I expected not to see a `no-loss-of-precision` issue reported for this number.

**What actually happened? Please include the actual, raw output from ESLint.**

An issue is reported for this rule.

**Are you willing to submit a pull request to fix this bug?**

Yes, if the rule creator won't do it themself 🙂

**Comment by anikethsaha** [comment]
I think it should be reported in `typescript-eslint` repo ?

cause [numeric-separator](https://github.com/tc39/proposal-numeric-separator) is still in Stage 3 and eslint doesnt support stage 3 proposal . 

also, the [compiled](https://typescript-play.js.org/#code/MYewdgzgLgBAHgLjAVwLYCMCmAnGBeGARgH0AGcgbiA) JS for this TS code is working [fine with eslint](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbG9zcy1vZi1wcmVjaXNpb246IFwiZXJyb3JcIiAqL1xuY29uc3QgeCA9IDEwMDA7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by kaicataldo** [comment]
As @anikethsaha mentions above, this is still currently experimental syntax. We will add support for this syntax in core rules once it reaches stage 4.

**Comment by merlinnot** [comment]
Thanks, I'll be waiting for it :)

**Comment by bricss** [comment]
Hi everyone,

I hit this issue today as well, and I can see potential problem in here.

Fan fact, is that [Numeric separators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Numeric_separators) is broadly used in many projects and supported already by ALL modern browsers and node as well, for around of one year. Plus, it doesn't have anything to do with TS, at all.

At the moment, it's impossible to use this new rule in any project that's already use separators in code.

It should be a way at least, to add an option to this rule to allow the character / code to be specified in numbers.

**Comment by kaicataldo** [comment]
While we can't implement support until this reaches stage 4, it might be worth checking out [`babel-eslint`](https://www.npmjs.com/package/babel-eslint) to see if that allows you to lint this syntax.

**Comment by bricss** [comment]
I have checked rule with [`babel-eslint`](https://www.npmjs.com/package/babel-eslint) and can confirm, that there is no luck. It still failing for every number with `_` in it. 😥

**Comment by mdjermanovic** [comment]
> I have checked rule with [`babel-eslint`](https://www.npmjs.com/package/babel-eslint) and can confirm, that there is no luck. It still failing for every number with `_` in it.

The core `no-loss-of-precision` rule will not work well with numeric separators until that syntax reaches stage 4 and we update the rule, regardless of the parser being used.

In the meantime, if you're using `@typescript-eslint/parser`, they have already implemented [`@typescript-eslint/no-loss-of-precision`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-loss-of-precision.md) rule which extends the core rule and works well with numeric separators.

If you're using `babel-eslint` parser, you could ask [`eslint-plugin-babel`](https://github.com/babel/eslint-plugin-babel) to extend the core rule in a similar way.

**Comment by mdjermanovic** [comment]
Numeric separators are stage 4 now. 

https://github.com/eslint/eslint/issues/13568 tracks everything that should be updated to support this syntax, and that includes `no-loss-of-precision` rule.

**Comment by mdjermanovic** [comment]
Fixed by #13574, ESLint v7.8.0
